packages/widget/src/components/common/dropdown/dropdown.ts

```diff
@@ -105,17 +110,31 @@ export class Dropdown extends BaseComponent {
     );
   }
 
-  _renderOptions(): Generator<unknown, void> {
+  _renderOptions(): Generator<unknown, void> | HTMLTemplateResult {
+    function renderOptionContent({
+      customOptionHtml,
+      name,
+      icon
+    }: DropdownOption): HTMLTemplateResult {
+      return when(
+        customOptionHtml,
+        () => customOptionHtml,
+        () => html`
+          ${icon || ''}
+          <span class="optionName">${capitalize(name)}</span>
+        `
+      ) as HTMLTemplateResult;
+    }
```

Comment:
i think @wainola mentioned it somewhere else, but i strongly suggest not declaring nested functions/methods. it is not/shouldn't be coupled with the method that's invoking it + when extracted, these functions/methods won't be redeclared on each `_renderOptions` call.